hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/HandleContext.java

```diff
@@ -358,65 +365,76 @@ default <T> T dispatchPrecedingTransaction(@NonNull TransactionBody txBody, @Non
      *
      * <p>A {@link TransactionCategory#PRECEDING}-transaction must not dispatch a child transaction.
      *
-     * @param txBody the {@link TransactionBody} of the child transaction to dispatch
+     * @param txBody             the {@link TransactionBody} of the child transaction to dispatch
      * @param recordBuilderClass the record builder class of the child transaction
-     * @param callback a {@link Predicate} callback function that will observe each primitive key
+     * @param callback           a {@link Predicate} callback function that will observe each primitive key
+     * @param syntheticPayer   the payer of the child transaction
      * @return the record builder of the child transaction
-     * @throws NullPointerException if any of the arguments is {@code null}
+     * @throws NullPointerException     if any of the arguments is {@code null}
      * @throws IllegalArgumentException if the current transaction is a
-     * {@link TransactionCategory#PRECEDING}-transaction or if the record builder type is unknown to the app
+     *                                  {@link TransactionCategory#PRECEDING}-transaction or if the record builder type is unknown to the app
      */
     @NonNull
     <T> T dispatchChildTransaction(
-            @NonNull TransactionBody txBody, @NonNull Class<T> recordBuilderClass, @NonNull Predicate<Key> callback);
+            @NonNull TransactionBody txBody,
+            @NonNull Class<T> recordBuilderClass,
+            @NonNull Predicate<Key> callback,
+            @NonNull AccountID syntheticPayer);
```

Comment:
I do not think this change is correct.  The payer should come from the transaction ID, as normal, when one is present.  The handlers depend (and should depend) on the presence of a transaction ID, so what the dispatcher should be doing, in my opinion, is read the payer from the transaction ID on the child transaction, and if that is null, then (and only then) apply a synthetic payer if provided (or error out if the payer is not provided).
Then we can overload this method with one version which does not provide a synthetic payer and one which does (informing the implementation up front that we do, or do not, intent to offer a synthetic payer).

Reply:
👍 Pushed commit adopting this overloaded `dispatch*()` approach.